### PR TITLE
expression: refine compare timestamp col with string constant

### PR DESCRIPTION
**Pull request by XuHuaiyu**
When comparing a timestamp column with a non-time constant,
they should be compared as timestamp rather than datetime since
timezone should be considered.

PTAL @winkyao @zz-jason

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -2705,6 +2705,12 @@ func (s *testIntegrationSuite) TestCompareBuiltin(c *C) {
 	tk.MustExec("create table t(a bigint unsigned)")
 	tk.MustExec("insert into t value(17666000000000000000)")
 	tk.MustQuery("select * from t where a = 17666000000000000000").Check(testkit.Rows("17666000000000000000"))
+
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a timestamp)")
+	tk.MustExec("insert into t value('1991-05-06 04:59:28')")
+	tk.MustExec("delete from t where a = '1991-05-06 04:59:28'")
```
Comment (winkyao) [inline expression/integration_test.go]: add `tk.MustQuery("admin check table t")` here.

Comment (XuHuaiyu) [reply expression/integration_test.go]: why we need this?

Comment (winkyao) [reply expression/integration_test.go]: Never mind.

**Comment by zz-jason** [comment]
@winkyao PTAL

**Comment by shenli** [comment]
LGTM

**Comment by crazycs520** [comment]
LGTM

**Pull request merged**